Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -268,31 +249,539 @@ def embProdEmbOfIsAlgebraic [Algebra E K] [IsScalarTower F E K] (halg : Algebra.
         (IsAlgClosure.equivOfAlgebraic E K (AlgebraicClosure K) (AlgebraicClosure E)
           halg).restrictScalars F).symm
 
-/-- If `K / E / F` is a field extension tower, such that `K / E` is algebraic, then
-$[E:F]_s [K:E]_s = [K:F]_s$. See also `lift_rank_mul_lift_rank`. -/
-theorem lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic
-    [Algebra E K] [IsScalarTower F E K] (halg : Algebra.IsAlgebraic E K) :
-    Cardinal.lift.{w} (sepDegree F E) * Cardinal.lift.{v} (sepDegree E K) =
-      Cardinal.lift.{v} (sepDegree F K) := by
-  have := (embProdEmbOfIsAlgebraic F E K halg).trans
-    (Equiv.ulift.{v} (α := Emb F K)).symm |>.cardinal_eq
-  simpa only [Cardinal.mk_prod, Cardinal.mk_uLift] using this
-
-/-- The same-universe version of `Field.lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic`.
-See also `rank_mul_rank`. -/
-theorem sepDegree_mul_sepDegree_of_isAlgebraic
-    (A : Type u) (B : Type u) [Field A] [Field B] [Algebra A B]
-    (C : Type u) [Field C] [Algebra A C] [Algebra B C]
-    [IsScalarTower A B C] (halg : Algebra.IsAlgebraic B C) :
-    sepDegree A B * sepDegree B C = sepDegree A C := by
-  simpa only [Cardinal.lift_id] using lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic A B C halg
-
-/-- The `Field.finSepDegree` version of `Field.lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic`.
-See also `FiniteDimensional.finrank_mul_finrank'`. -/
+/-- If `K / E / F` is a field extension tower, such that `K / E` is algebraic, then their
+(finite) separable degree satisfies the tower law
+$[E:F]_s [K:E]_s = [K:F]_s$. See also `FiniteDimensional.finrank_mul_finrank'`. -/
 theorem finSepDegree_mul_finSepDegree_of_isAlgebraic
     [Algebra E K] [IsScalarTower F E K] (halg : Algebra.IsAlgebraic E K) :
     finSepDegree F E * finSepDegree E K = finSepDegree F K := by
-  simpa only [Cardinal.toNat_mul, Cardinal.toNat_lift] using congr_arg Cardinal.toNat
-    (lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic F E K halg)
+  simpa only [Nat.card_prod] using Nat.card_congr (embProdEmbOfIsAlgebraic F E K halg)
+
+end Field
+
+namespace Polynomial
+
+variable {F E}
+
+variable (f : F[X])
+
+/-- If a polynomial has all roots over a field, then it has no repeated roots on that field
+if and only if it is separable. -/
+theorem nodup_roots_iff_of_splits (hf : f ≠ 0) (h : f.Splits (RingHom.id F)) :
```

Comment:
Can this go the file where `Polynomial.Separable` is defined?

Reply:
Sure, I'll check later.